[BUG] Unable to access schema properties in Document object in TypeScript

<!-- Not filling out ALL of the relevant fields in this issue will cause your issue to be closed -->

### Summary:
Tested with a basic schema (code provided below).

While trying to access a property of document, TS complains with following error:
```
⨯ Unable to compile TypeScript:
src/snippet.ts:49:19 - error TS2339: Property 'pk' does not exist on type 'Document'.
```


### Code sample:

#### General

Running the following ts script results in compilation error.

```
import * as dynamoose from "dynamoose";

dynamoose.aws.sdk.config.update({
  region: "ap-southeast-2"
});

// use the dynamodb local
dynamoose.aws.ddb.local();

const EntrySchema = new dynamoose.Schema(
  {
    pk: {
      type: String,
      hashKey: true
    },
    sk: {
      type: String,
      rangeKey: true
    },
    entries: {
      type: Array,
      schema: [String]
    }
  },
  {
    saveUnknown: false,
    timestamps: true
  }
);

const Entry = dynamoose.model("table_name", EntrySchema, {
  create: true
});

Entry.create({
  pk: "4545",
  sk: "abc#44",
  entries: ["hello", "world"]
}).then(doc => {
  console.log(doc.pk); // TS compile error: src/snippet.ts:49:19 - error TS2339: Property 'pk' does not exist on type 'Document'.
});

```


### Current output and behavior (including stack trace):
 TS compile error: src/snippet.ts:49:19 - error TS2339: Property 'pk' does not exist on type 'Document'.


### Expected output and behavior:
Should just work.


### Environment:

Operating System: Mac
Operating System Version: Catalina
Node.js version (`node -v`): 12.16
NPM version: (`npm -v`): 6.14
Dynamoose version: 2.3


### Other information (if applicable):




### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

--- a/test/types/model/Create.ts
+++ b/test/types/model/Create.ts
@@ -3,6 +3,7 @@
 import {UserTypedModel} from "../Model";
 
 const shouldCreateSuccessfully = UserTypedModel.create({"id": "1", "name": "john", "age": 25});
+const shouldAccessPropertySuccessfully = (await shouldCreateSuccessfully).id;
 
 const shouldNotFailWithNotAllAttributesPassedIn = UserTypedModel.create({"id": "1"});
 

--- a/test/types/tsconfig.json
+++ b/test/types/tsconfig.json
@@ -1,9 +1,9 @@
 {
 	"compilerOptions": {
-		"module": "commonjs",
+		"module": "esnext",
 		"moduleResolution": "node",
 		"lib": ["es2015"],
-		"target": "es6",
+		"target": "es2017",
 		"noEmit": true
 	},
 	"include": ["./**/*"]

